### PR TITLE
Converted the project to cljc

### LINKED ISSUE #37: Convert CTIM to cljc
Convert the whole project to cljc so it can be used from clojurescript

### PR DESCRIPTION
**Pull request by gbuisson**
closes #37 

- Tests are also converted run them with either 
`lein doo node node once`or `lein doo phantomjs test once`
-  A quick cljs sample will be added later

### CODE REVIEW

**Comment by yogsototh** [comment]
LGTM if you squash all commits!

File: src/ctim/generators/id.cljc [hunk 1]
```diff
@@ -17,18 +19,27 @@
...
 (def gen-path
-  (chuck/string-from-regex
-   #"(\/([\w][-\w]{0,5}(\.[-\w]{1,4}){0,3})){1,3}"))
+  #?(:clj (chuck/string-from-regex
+           #"(\/([\w][-\w]{0,5}(\.[-\w]{1,4}){0,3})){1,3}")
+     :cljs
+     (gen/fmap (fn [[f s]]
+                 (str "/" f "/" s))
+               (gen/tuple (gen-str-3+ gen-char-alpha-lower)
+                          (gen-str-3+ gen-char-alpha-lower)))))
```
Comment (polygloton) [inline src/ctim/generators/id.cljc]: I think you should just use the :cljs implementation for both here.  Maybe we can get rid of chuck.

File: src/ctim/lib/time.cljc [hunk 2]
```diff
@@ -41,21 +51,15 @@
...
   (time/after? (coerce-to-datetime left)
                (coerce-to-datetime right)))
 
-(defn sql-now []
-  (coerce-to-sql-time (now)))
-
-(def to-sql-time coerce-to-sql-time)
-
-(def from-sql-time coerce-to-date)
-
```
Comment (polygloton) [inline src/ctim/lib/time.cljc]: Why are these deleted?  Are they not used?

Comment (gbuisson) [inline src/ctim/lib/time.cljc]: Yes since we removed sql support there is no need for this anymore.
also, as suggested previously we should split this code into a common toolset lib shared by our projects.
